src/Package/Impl/ProjectSystem/RDebugLaunchProvider.cs

```diff
@@ -85,31 +87,126 @@ public RDebugLaunchProvider(ConfiguredProject configuredProject, IRInteractiveWo
+            bool transferFiles = await _properties.GetTransferProjectOnRunAsync();
+            string remotePath = await _properties.GetRemoteProjectPathAsync();
+            string filterString = await _properties.GetFileFilterAsync();
+
+            var activeProject = _pss.GetActiveProject();
+            if (transferFiles && Session.IsRemoteSession && activeProject != null) {
+                await SendProjectAsync(activeProject, remotePath, filterString);
+            }
+
+            // user must set the path for local or remote cases
+            var startupFile = await GetStartupFileAsync(transferFiles, activeProject);
+
+            if (string.IsNullOrWhiteSpace(startupFile)) {
                 _interactiveWorkflow.ActiveWindow?.InteractiveWindow.WriteErrorLine(Resources.Launch_NoStartupFile);
                 return;
             }
-            await SourceFileAsync(startupFile, Invariant($"{Resources.Launch_StartupFileDoesNotExist} {startupFile}"));
+            await SourceFileAsync(transferFiles, startupFile, Invariant($"{Resources.Launch_StartupFileDoesNotExist} {startupFile}"));
 
             var settingsFile = await _properties.GetSettingsFileAsync();
             if (!string.IsNullOrWhiteSpace(settingsFile)) {
-                var activeProject = _pss.GetActiveProject();
                 if (activeProject != null) {
-                    settingsFile = settingsFile.MakeAbsolutePathFromRRelative(Path.GetDirectoryName(activeProject.FullName));
+                    var dirPath = Path.GetDirectoryName(activeProject.FullName);
+                    settingsFile = settingsFile.MakeAbsolutePathFromRRelative(dirPath);
                     if (FileSystem.FileExists(settingsFile)) {
-                        await SourceFileAsync(settingsFile, Invariant($"{Resources.Launch_SettingsFileDoesNotExist} {settingsFile}"));
+                        if (Session.IsRemoteSession) {
+                            var remoteSettingsPath = GetRemoteSettingsFile(settingsFile, dirPath, remotePath);
+                            await SourceFileAsync(transferFiles, remoteSettingsPath, Invariant($"{Resources.Launch_SettingsFileDoesNotExist} {settingsFile}"));
+                        } else {
+                            await SourceFileAsync(transferFiles, settingsFile, Invariant($"{Resources.Launch_SettingsFileDoesNotExist} {settingsFile}"));
+                        }
                     }
                 }
             }
         }
 
-        private async Task SourceFileAsync(string file, string errorMessage) {
-            if (!FileSystem.FileExists(file)) {
+        private async Task SourceFileAsync(bool transferFiles, string file, string errorMessage) {
+            bool fExists = false;
+            if (transferFiles && Session.IsRemoteSession) {
+                fExists = await Session.EvaluateAsync<bool>(Invariant($"file.exists({file.ToRPath().ToRStringLiteral()})"), REvaluationKind.Normal);
```

Comment:
No need for `Invariant` - `EvaluateAsync` provides overloads that take `FormattableString`, and will do this for you.